osquery/utils/windows/shellitem.cpp

```diff
@@ -0,0 +1,471 @@
+                                       "01A3057A-74D6-4E80-BEA7-DC4C212CE50A",
+                                       "46588AE2-4CBC-4338-BBFC-139326986DCE",
+                                       "4D545058-4FCE-4578-95C8-8698A9BC0F49",
+                                       "56A3372E-CE9C-11D2-9F0E-006097C686F6",
+                                       "6444048F-4C8B-11D1-8B70-080036B11A03",
+                                       "64440490-4C8B-11D1-8B70-080036B11A03",
+                                       "64440491-4C8B-11D1-8B70-080036B11A03",
+                                       "64440492-4C8B-11D1-8B70-080036B11A03",
+                                       "8F052D93-ABCA-4FC5-A5AC-B01DF4DBE598",
+                                       "B725F130-47EF-101A-A5F1-02608C9EEBAC",
+                                       "D5CDD502-2E9C-101B-9397-08002B2CF9AE",
+                                       "D5CDD505-2E9C-101B-9397-08002B2CF9AE",
+                                       "EF6B490D-5CD8-437A-AFFC-DA8B60EE4A3C",
+                                       "F29F85E0-4FF9-1068-AB91-08002B27B3D9"};
+namespace osquery {
+std::string guidParse(const std::string& guid_little) {
+  std::vector<std::string> guids;
+  guids.push_back(guid_little.substr(0, 8));
+  guids.push_back(guid_little.substr(8, 4));
+  guids.push_back(guid_little.substr(12, 4));
+
+  std::string guid_4 = guid_little.substr(16, 4);
+  std::string guid_5 = guid_little.substr(20, 12);
+
+  // The first 16 GUID characters are in litte endian format
+  for (auto& guid : guids) {
+    std::reverse(guid.begin(), guid.end());
+    for (std::size_t i = 0; i < guid.length(); i += 2) {
+      std::swap(guid[i], guid[i + 1]);
+    }
+  }
+  std::string guid_string =
+      guids[0] + "-" + guids[1] + "-" + guids[2] + "-" + guid_4 + "-" + guid_5;
+  return guid_string;
+}
+
+ShellFileEntryData fileEntry(const std::string& shell_data) {
+  size_t offset;
+  std::string extension_sig;
+  size_t entry_offset = 0;
+  // Find "0400EFBE" offset
+  if (shell_data.find("0400EFBE") != std::string::npos) {
+    offset = shell_data.find("0400EFBE");
+    extension_sig = shell_data.substr(offset, 8);
+    entry_offset = offset - 8;
+  }
+  ShellFileEntryData file_entry;
+
+  if (entry_offset == 0) {
+    LOG(WARNING)
+        << "Could not find supported file entry extension in shell data: "
+        << shell_data;
+    file_entry.path = "[UNSUPPORTED SHELL EXTENSION]\\";
+    return file_entry;
+  }
+
+  std::string version = shell_data.substr(entry_offset + 4, 4);
+  // swap endianess
+  std::reverse(version.begin(), version.end());
+  for (std::size_t i = 0; i < version.length(); i += 2) {
+    std::swap(version[i], version[i + 1]);
+  }
+  file_entry.version = std::stoi(version, nullptr, 16);
+  if (file_entry.version < 7 || file_entry.version == 1) {
+    LOG(WARNING)
+        << "Shellitem format unsupported. Expecting version 1 or version 7 or "
+           "higher: "
+        << shell_data;
```

Comment:
This looks perhaps not as intended as the second case will never hit

Reply:
good catch, removed old logic used when first researching shellbags